mountpoint-s3-fs/src/manifest/csv_reader.rs

```diff
@@ -7,26 +8,65 @@ use super::builder::InputManifestEntry;
 
 pub struct CsvReader<R: Read> {
     reader: DeserializeRecordsIntoIter<R, CsvEntry>,
+    file_path: String,
+    hasher: Option<Hasher>,
+    expected_checksum: u32,
 }
 
 impl<R: Read> CsvReader<R> {
-    pub fn new(reader: R) -> Self {
+    pub fn new(reader: R, file_path: &str, expected_checksum: u32) -> Self {
         let reader = ReaderBuilder::new()
             .has_headers(false)
             .from_reader(reader)
             .into_deserialize();
-        Self { reader }
+        Self {
+            reader,
+            file_path: file_path.to_string(),
+            hasher: Some(Hasher::new()),
+            expected_checksum,
+        }
+    }
+
+    fn update_running_checksum(&mut self, csv_entry: &CsvEntry) {
+        // we re-compute checksum of each entry, because `Hasher` doesn't support combining pre-computed checksums
+        if let Some(hasher) = self.hasher.as_mut() {
+            hasher.update(csv_entry.partial_key.as_bytes());
+            hasher.update(csv_entry.etag.as_bytes());
+            // we encode size with big endian byte order and with a fixed width of 8 bytes (rust: u64, java: long)
```

Comment:
Should we make `csv_entry.size` a `u64` then?

Reply:
`InodeStat::size` is `usize`, I think the best we can do is to fail on CSV parsing on machines where `usize` is less than 8 bytes and larger size was found in the manifest (that's how it works now)

Reply:
I think we should use `u64` to match the checksum calculation and then validate when building the manifest (e.g. CsvEntry -> InputManifestEntry)